core/shared/src/main/scala/scodec/bits/ByteVector.scala

```diff
@@ -1218,6 +1233,13 @@ object ByteVector {
         i += 1
       }
     }
+    def copyToBuffer(buffer: ByteBuffer, offset: Long, size: Int): Unit = {
+      var i = 0
+      while (i < size && buffer.remaining > 0) {
```

Comment:
I see.
It's not that crucial anyway because the `ByteBuffer.put(array, offset, length)` overload internally loops as well (rather than calling `System.arrayCopy`), so the benefit isn't that great.